homeassistant/components/flickelectric/sensor.py

```diff
@@ -0,0 +1,174 @@
+"""Support for Flick Electric Pricing data."""
+import asyncio
+from datetime import datetime as dt, timedelta
+import logging
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_FRIENDLY_NAME,
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+_AUTH_URL = "https://api.flick.energy/identity/oauth/token"
+_RESOURCE = "https://api.flick.energy/customer/mobile_provider/price"
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+_TOKEN_REFRESH_INTERVAL = timedelta(days=1)
+
+ATTRIBUTION = "Data provided by Flick Electric"
+FRIENDLY_NAME = "Flick Power Price"
+UNIT_NAME = "cents"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Config Flow added, let me know of any suggestions/improvements to make to it.

Reply:
We can remove the platform schema. :+1:

Reply:
Done :)